src/kOS.Safe/Encapsulation/QueueValue.cs

```diff
@@ -13,71 +13,59 @@ public QueueValue() : this(new Queue<T>())
         {
         }
 
-        public QueueValue(IEnumerable<T> queueValue) : base(new Queue<T>(queueValue))
+        public QueueValue(IEnumerable<T> queueValue) : base("QUEUE", new Queue<T>(queueValue))
         {
             QueueInitializeSuffixes();
         }
 
-        public override int Count
-        {
-            get { return Collection.Count; }
-        }
-
         public T Pop()
         {
-            return Collection.Dequeue();
+            return InnerEnumerable.Dequeue();
         }
 
         public void Push(T val)
         {
-            Collection.Enqueue(val);
+            InnerEnumerable.Enqueue(val);
         }
 
         public override Dump Dump()
         {
             var result = new DumpWithHeader
             {
-                Header = "QUEUE of " + Collection.Count() + " items:"
+                Header = "QUEUE of " + InnerEnumerable.Count() + " items:"
```

Comment:
Hmm. Maybe get the name from KOSNomenclature.GetKOSName(GetType()) instead of hardcoding the word "QUEUE" here, thus putting all the configuration of what we call this thing when talking to the user into one single place - the attribute.  

Come to think of it, we could add that as a get property to the base class `Structure`, so all Structure's can just use KOSName as a property.